projectboard/main.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright © 2023 The Kubernetes Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/google/go-github/v48/github"
+	githubql "github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+)
+
+const (
+	// perPage is the number of items to return per page.
+	perPage = 100
+	// tokenEnv is the environment variable that contains the
+	// GitHub token to use for authentication.
+	// GITHUB_TOKEN is a personal access token with the following scopes:
+	// - repo (all)
+	// - read:org
+	// - project (all)
+	tokenEnv = "GITHUB_TOKEN"
+	// orgNameEnv is the environment variable that contains the
+	// name of the GitHub organization to query.
+	// e.g. kubernetes
+	orgNameEnv = "GITHUB_ORG"
+	// projectNameEnv is the environment variable that contains the
+	// name of the GitHub project to query.
+	// e.g. SIG Auth
+	projectNameEnv = "GITHUB_PROJECT"
+	// filterLabelEnv is the environment variable that contains the
+	// label to filter issues and PRs by.
+	// If empty, all issues and PRs will be added to the project.
+	// e.g. sig/auth
+	filterLabelEnv = "FILTER_LABEL"
+)
+
+type ghClient struct {
+	*github.Client
+	v4Client *githubql.Client
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancel()
+
+	orgName := os.Getenv(orgNameEnv)
+	if orgName == "" {
+		log.Fatalf("environment variable %q must be set", orgNameEnv)
+	}
+	projectName := os.Getenv(projectNameEnv)
+	if projectName == "" {
+		log.Fatalf("environment variable %q must be set", projectNameEnv)
+	}
+	filterLabel := os.Getenv(filterLabelEnv)
+
+	token := os.Getenv(tokenEnv)
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := ghClient{Client: github.NewClient(tc), v4Client: githubql.NewClient(tc)}
+
+	projectID, err := client.getProjectID(ctx, orgName, projectName)
+	log.Fatal(err)
+
+	repos, err := client.listRepos(ctx, orgName)
```

Comment:
How about we add another ENV to provide a short "list of repos" to search on?

If the ENV is not set, fallback to the current implementation (i.e., fetching issues/PRs from all repos in the given organisation.)